lib/vre/ch31_form.rb

```diff
@@ -63,49 +63,71 @@ def format_payload_for_vre
 
-      {
-        veteranAddress: {
-          isForeign: vet_address['country'] != 'USA',
-          isMilitary: vet_address['isMilitary'] || false,
-          countryName: vet_address['country'],
-          addressLine1: vet_address['street'],
-          addressLine2: vet_address['street2'],
-          addressLine3: vet_address['street3'],
-          city: vet_address['city'],
-          stateCode: vet_address['state'],
-          zipCode: vet_address['postalCode']
-        }
+      adjusted_address = {
+        veteranAddress: vet_address
       }
+
+      return adjusted_address if adjusted_address.dig(:veteranAddress, :isForeign) == false
+
+      international_address = adjusted_address[:veteranAddress]
+      international_address[:internationPostalCode] = international_address.delete(:zipCode)
+      international_address[:province] = international_address.delete(:stateCode)
+
+      adjusted_address
     end
 
     def claim_form_hash
       @claim.parsed_form
     end
 
+    def adjusted_veteran_information
+      vet_info = claim_form_hash['veteranInformation'].with_indifferent_access
+
+      vet_info['VAFileNumber'] = vet_info.delete('vaFileNumber') if vet_info.key?('vaFileNumber')
+
+      vet_info['VAFileNumber'] = vet_info['ssn'] if vet_info['VAFileNumber'].nil?
+
+      vet_info
+    end
+
     def new_address
-      new_address = claim_form_hash['newAddress']
-      {
-        "newAddress": {
-          "isForeign": new_address['country'] != 'USA',
-          "isMilitary": new_address['isMilitary'],
-          "countryName": new_address['country'],
-          "addressLine1": new_address['street'],
-          "addressLine2": new_address['street2'],
-          "addressLine3": new_address['street3'],
-          "city": new_address['city'],
-          "province": new_address['state'],
-          "internationalPostalCode": new_address['postalCode']
-        }
+      new_address = mapped_address_hash(claim_form_hash['newAddress'])
+
+      adjusted_new_address = {
+        newAddress: new_address
       }
+
+      return adjusted_new_address if adjusted_new_address.dig(:newAddress, :isForeign) == false
+
+      new_vet_international_address = adjusted_new_address[:newAddress]
+
+      new_vet_international_address[:internationalPostalCode] = new_vet_international_address.delete(:zipCode)
+      new_vet_international_address[:province] = new_vet_international_address.delete(:stateCode)
+
+      adjusted_new_address
+    end
+
+    def mapped_address_hash(client_hash)
+      {
+        isForeign: client_hash['country'] != 'USA',
+        isMilitary: client_hash['isMilitary'] || false,
+        countryName: client_hash['country'],
+        addressLine1: client_hash['street'],
+        addressLine2: client_hash['street2'],
+        addressLine3: client_hash['street3'],
+        city: client_hash['city'],
+        stateCode: client_hash['state'],
+        zipCode: client_hash['postalCode']
+      }.with_indifferent_access
```

Comment:
what does indiffernt access get you?  looks like you're using symbol keys everywhere

Reply:
Good callout. Just being overly cautious